core-bundle/src/Image/Preview/ImaginePreviewProvider.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Image\Preview;
+
+use Contao\Image\ImageDimensions;
+use Imagine\Factory\ClassFactoryInterface;
+use Imagine\Gd\Imagine as GdImagine;
+use Imagine\Gmagick\Imagine as GmagickImagine;
+use Imagine\Image\Box;
+use Imagine\Image\BoxInterface;
+use Imagine\Image\ImageInterface;
+use Imagine\Image\ImagineInterface;
+use Imagine\Image\Palette\CMYK;
+use Imagine\Image\Palette\Grayscale;
+use Imagine\Image\Palette\RGB;
+use Imagine\Imagick\Imagine as ImagickImagine;
+
+class ImaginePreviewProvider implements PreviewProviderInterface
+{
+    private ImagineInterface $imagine;
+
+    public function __construct(ImagineInterface $imagine)
+    {
+        $this->imagine = $imagine;
+    }
+
+    public function getFileHeaderSize(): int
+    {
+        return 0;
+    }
+
+    public function supports(string $path, string $fileHeader = '', array $options = []): bool
+    {
+        $format = strtolower(pathinfo($path, PATHINFO_EXTENSION));
+
+        return $this->imagineSupportsFormat($format);
```

Comment:
This method throws an "Unsupported Imagine implementation" exception. Should we handle it or let it bubble up?

Reply:
I think we should let it bubble up because it would be some sort of a misconfiguration.